threema-connector-test/src_test/threema/connector/test/process/GetReceiverInfoTest.java

```diff
@@ -0,0 +1,132 @@
+		recDatMail.setType(LookupType.EMAIL);
+		
+		ExecutionResult resultMail = bpmClient.start().subProcess(callable).execute(recDatMail);
+		receiverData resultDataMail = resultMail.data().last();		
+		History historyMail = resultMail.history();
+
+		assertThat(resultDataMail.getApiResponse()).contains("404");
+		assertThat(historyMail.elementNames()).contains("call(receiverData)");
+		assertThat(historyMail.elementNames()).contains("LookupId");
+		assertThat(historyMail.elementNames()).doesNotContain("LookupPubKey");
+		
+	}
+	
+	@Test
+	void getIDByValidPhone(BpmClient bpmClient) {
+		BpmElement callable = RECEIVER_INFO_PROCESS.elementName("call(receiverData)");
+		
+		String phone = VALID_ID;
+		receiverData recDatMail = new receiverData();
+		recDatMail.setIdentifier(phone);
+		recDatMail.setType(LookupType.PHONE);
+		
+		ExecutionResult resultMail = bpmClient.start().subProcess(callable).execute(recDatMail);
+		receiverData resultDataMail = resultMail.data().last();		
+		History historyMail = resultMail.history();
+
+		assertThat(resultDataMail.getApiResponse()).contains("200");
+		assertThat(historyMail.elementNames()).contains("call(receiverData)");
+		assertThat(historyMail.elementNames()).contains("LookupId");
+		assertThat(historyMail.elementNames()).contains("LookupPubKey");
+		
+	}
+	
+	@Test
+	void getPublicKeyByID(BpmClient bpmClient) {
+		BpmElement callable = RECEIVER_INFO_PROCESS.elementName("call(receiverData)");
+		
+		String threemaId = "validId";
+		receiverData recDatId = new receiverData();
+		recDatId.setIdentifier(threemaId);
+		recDatId.setType(LookupType.THREEMAID);
+		
+		ExecutionResult resultId = bpmClient.start().subProcess(callable).execute(recDatId);
+		receiverData resultDataId = resultId.data().last();
+		History historyId = resultId.history();
+		
+		assertThat(resultDataId.getApiResponse()).contains("200");
+		assertThat(resultDataId.getPublicKey()).isEqualTo("ffbb40cfced42f75c4d83c7d35300c0698bf3ef1ab49ace323a1bbc38ee23f36");
+		assertThat(historyId.elementNames()).contains("call(receiverData)");
+		assertThat(historyId.elementNames()).contains("LookupPubKey");
+		assertThat(historyId.elementNames()).doesNotContain("LookupId");
+	}
+	
+	
+	@Test
+	void getPublicKeyByInvalidID(BpmClient bpmClient) {
+		BpmElement callable = RECEIVER_INFO_PROCESS.elementName("call(receiverData)");
+		
+		String threemaId = "invalidID";
+		receiverData recDatId = new receiverData();
+		recDatId.setIdentifier(threemaId);
+		recDatId.setType(LookupType.THREEMAID);
+		
+		ExecutionResult resultId = bpmClient.start().subProcess(callable).execute(recDatId);
+		receiverData resultDataId = resultId.data().last();
+		History historyId = resultId.history();
+		
+		
+		assertThat(resultDataId.getApiResponse()).contains("404");
+		assertThat(historyId.elementNames()).contains("call(receiverData)");
+		assertThat(historyId.elementNames()).contains("LookupPubKey");
+		assertThat(historyId.elementNames()).doesNotContain("LookupId");
```

Comment:
I'd cluster assertions on the same object into one to increase the readability
```
assertThat(historyId.elementNames())
  .contains("call(receiverData)")
  .contains("LookupPubKey")
  .doesNotContain("LookupId");
```

Reply:
implemented suggested change